flutter/flutter-provider/state-management-in-flutter-using-provider-Part2.md

```diff
@@ -0,0 +1,209 @@
+we have discussed the problem of using `setState` to update the app's state in the moderate or complex app.
+
+## What we are going to know in this article
+
+1. Why we should use provider
+2. What are the types of provider
+3. How to use provider in our code
+
+In this article we'll know about `provider` package owned by  [Remi Rousselet](https://github.com/rrousselGit/provider), this is one of the best state management technique we can use in our app.
+
+### Why we should use provider
+
+ On the beginner stage, we generally use the `setState` to update the state of our app, because its easy to do so, and on this stage, if we try to use some moderate or say a complex type of state management technique like `flutter_bloc` or `Mobx`, it is not easy for beginners to take a big jump from using `setState` to these state management technique 😓
+
+ 
+``` 
+provider uses InheritedWidget internally, to manage all those complex computations and provide us simple and great feature to manage our state
+```
+
+&nbsp;
+
+> Why `provider`? We have some other techniques that can pass data in the widget tree.
+
+Yes, there is some other way, such as passing data via constructors this can be done, but definitely not the good approach.
+
+***Explanation***
+>Passing data via constructors, this is not feasible if data is 3 or 4 widgets above in the tree and we are trying to pass the data via constructor through a widget, and in doing so, all those widgets will rebuild just for passing the data. This may cause the performance issue and our code get too much complicated.
+
+So, what we should do now? 
+
+We will use provider 
+
+``` 
+flutter_bloc and Mobx use provider internally 🥳 And now the flutter team also supports to use provider for state management.
+```
+
+So, let's get started knowing more about provider
+
+### What are the types of provider
+
+&nbsp;
+
+Provider has enough categories to handle different type of objects
+
+The complete list of all the objects available is [here](https://pub.dev/documentation/provider/latest/provider/provider-library.html)
+
+| Name|Description|
+|-----|-----------|
+|Provider|The most basic form of provider. It takes a value and exposes it, whatever the value is.|
+|ListenableProvider|A specific provider for Listenable object. ListenableProvider will listen to the object and ask widgets which depend on it to rebuild whenever the listener is called.|
+|ChangeNotifierProvider|A specification of ListenableProvider for ChangeNotifier. It will automatically call `ChangeNotifier.dispose` when needed.|
+|ValueListenableProvider|Listen to a ValueListenable and only expose `ValueListenable.value`.|
+|StreamProvider|Listen to a Stream and expose the latest value emitted.|
+|FutureProvider|Takes a `Future` and updates dependents when the future completes.|
+
+&nbsp;
+
+### So, let's discover more about them
+
+### **Provider**
```

Comment:
This is inside `###` heading, hence it should be `####`. 
Headings are bold by default, you don't have to use **